src/thunderbots/software/ai/passing/evaluation.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Implementation of evaluation functions for passing
+ */
+
+// TODO: declare things const& and whatnot to maybe improve speed
+
+#include "ai/passing/evaluation.h"
+
+#include "util/parameter/dynamic_parameters.h"
+
+using namespace AI::Passing;
+
+double AI::Passing::getStaticPositionQuality(const Field& field, const Point& position)
+{
+    // This constant is used to determine how steep the sigmoid slopes below are
+    static const double sigmoid_width = 0.1;
+
+    // The offset from the sides of the field for the center of the sigmoid functions
+    double x_offset =
+        Util::DynamicParameters::AI::Passing::static_position_quality_x_offset.value();
+    double y_offset =
+        Util::DynamicParameters::AI::Passing::static_position_quality_y_offset.value();
+    double friendly_goal_weight =
+        Util::DynamicParameters::AI::Passing::
+            static_position_quality_friendly_goal_distance_weight.value();
+
+    // Make a slightly smaller field, and positive weight values in this reduced field
+    double half_field_length = field.length() / 2;
+    double half_field_width  = field.width() / 2;
+    Rectangle reduced_size_field(
+        Point(-half_field_length + x_offset, -half_field_width + y_offset),
+        Point(half_field_length - x_offset, half_field_width - y_offset));
+    double on_field_quality =
+        rectangleSigmoid(reduced_size_field, position, sigmoid_width);
+
+    // Add a negative weight for positions closer to our goal
+    Vector vec_to_friendly_goal      = Vector(field.friendlyGoal().x() - position.x(),
+                                         field.friendlyGoal().y() - position.y());
+    double distance_to_friendly_goal = vec_to_friendly_goal.len();
+    double near_friendly_goal_quality =
+        (1 -
+         std::exp(-friendly_goal_weight * (std::pow(5, -2 + distance_to_friendly_goal))));
+
+    // Add a strong negative weight for positions within the enemy defense area, as we
+    // cannot pass there
+    double in_enemy_defense_area_quality =
+        1 - rectangleSigmoid(field.enemyDefenseArea(), position, 0.1);
```

Comment:
Should this `0.1` be the `sigmoid_width` variable defined above?

Reply:
yup, good catch. Changed